### ISSUE TITLE
Unity screen turning white on scene load (Android)

### ISSUE THREAD

**Issue by AkhilRaja** [juicycleff/flutter-unity-view-widget#578]
**Describe the bug**
Unity Scene is loaded and the scripts are all running in the background, but just for the first launch the screen is white until it is unloaded and loaded. 

**To Reproduce**
1. Load the unity module for the first time for an AR scene
2. You will see a white screen instead of the camera opening up.

**Unity (please complete the following information):**
 - OS: iOS
 - Version [e.g. 2022.1.01f]

**Smartphone (please complete the following information):**
 - Device: iPhone 13 mini

**Comment by AkhilRaja** [comment]
@juicycleff also to add, it happens on the first time of every new app session. From the second time onward it works well.

**Comment by juicycleff** [comment]
Will take a look at it, just for now load unity widget in your splash screen widget if you have one with opacity set to 0 until I get back to you on this issue

**Comment by AkhilRaja** [comment]
Thankyou for checking,  no I don't have a splash screen. It's the default made with unity screen that's causing the trouble I believe. 

Let me know when you get a chance to take a look.

**Comment by Anishishi** [comment]
There seems to be a problem with the flutter-unity-widget package itself. Fixing the swift file as described in this comment could solve the problem.
https://github.com/juicycleff/flutter-unity-view-widget/issues/540#issuecomment-1063408129

**Comment by AkhilRaja** [comment]
@Anishishi I can confrim, that comment fixes it. I think the completed(rootview) is not geting called from the main UI thread. Just adding that fixes it. 
IMO the unity warming up logic isnt working well and needs to be refactored in the plugin

**Comment by juicycleff** [comment]
I agree, will be a small fix, working on windows atm, for now just unload unity in a splash screen, while I wrap up windows support

**Comment by dawiddszewczyk** [comment]
@juicycleff  @Ahmadre Yeah this issue is still on iOS on android everything works well.

**Comment by behnamsattar** [comment]
I was able to fix this by adding a manual unload and reload. Here is the code I'm using to solve this:

```dart
UnityWidgetController? _unityWidgetController;
  Timer? timer;
  bool _isCheckingLoadState = false;
  bool _hasBeenRestarted = false;

  void _onUnityCreated(UnityWidgetController controller) async {
    _unityWidgetController = controller;
    // Due to an issue with the Unity plugin when it sometimes fails to load the
    // Unity scene, we perform a reload after the first unity scene creation. It
    // ensures that (1) the scene is loaded correctly and (2) the scene restarts
    // after a flutter restart during debugging.
    if (!_hasBeenRestarted) {
      _hasBeenRestarted = true;
      await _reloadUnityScene();
    }
  }

  Future<void> _reloadUnityScene() async {
    await _unityWidgetController?.unload();
    timer = Timer.periodic(
      const Duration(milliseconds: 500),
      (_) async {
        if (!_isCheckingLoadState) {
          _isCheckingLoadState = true;
          final isLoaded = await _unityWidgetController?.isLoaded() ?? false;
          if (!isLoaded) {
            timer?.cancel();
            await _unityWidgetController?.create();
          }
          _isCheckingLoadState = false;
        }
      },
    );
  }
```

Please note that, at the moment, I'm getting an exception when calling the create method. I have filed an issue regarding this, which I hope to get resolved soon. The fix works as expected, though.

**Comment by kamami** [comment]
Same issue with newest release on flutter master
